fix: special shortcut for Windows live notifications

8thony opened the pull request:
Fixes #5954

The shortcut is created by default when the first notification fires, unless one of the `portable` or `externally-packaged` (new mode) modes are enabled.

`externally-packaged` is a new mode with this released, and can be useful when packaging Chatterino (e.g. to SCOOP)

pajlada commented on .CI/chatterino-installer.iss:
```diff
@@ -77,7 +77,7 @@ Source: "{#WORKING_DIR}vc_redist.x64.exe"; DestDir: "{tmp}"; Tasks: vcredist;
 
 [Icons]
-Name: "{autoprograms}\{#MyAppName}"; Filename: "{app}\{#MyAppExeName}"
+Name: "{autoprograms}\{#MyAppName}"; Filename: "{app}\{#MyAppExeName}"; AppUserModelID: ".Chatterino 2"
```
Is `AppUserModelID` the string that appears on top of the notification?

8thony replied:
No, it is used to associate the shortcut with the app.
I probably add it to the Desktop Shortcut as well.

8thony commented:
@Nerixyz there might be a bug with pinning c2 to the taskbar.

pajlada commented on src/widgets/settingspages/NotificationPage.cpp:
```diff
@@ -51,6 +51,14 @@ NotificationPage::NotificationPage()
                     "Show notification", getSettings()->notificationToast));
 #endif
 #ifdef Q_OS_WIN
+                settings.append(this->createCheckBox(
+                    "Create start menu shortcut (requires "
+                    "restart)",
+                    getSettings()->createShortcutForToasts,
+                    "When enabled, a shortcut will be created inside your "
+                    "start menu folder if needed by live notifications."
+                    "\n(On portable mode, this is disabled by "
+                    "default)"));
```
You can use `SettingWidget` here, which means you wouldn't have to update the `createCheckBox` implementation - see https://github.com/Chatterino/chatterino2/blob/7fee3f7e375a099b5207c215389343f8f1d95bf2/src/widgets/settingspages/GeneralPage.cpp#L1231-L1235 as a reference

pajlada replied:
nvm all good, you can't do that here yet

pajlada commented on src/singletons/Settings.hpp:
```diff
@@ -526,6 +527,9 @@ class Settings
     BoolSetting notificationToast = {"/notifications/enableToast", false};
+    BoolSetting createShortcutForToasts = {
+        "/notifications/createShortcutForToasts",
+        Modes::instance().isPortable ? false : true};
```
add a , at the end of the ternary, that way it formats a bit nicer, causing nicer diffs in the future if we have to edit it

pajlada reviewed (commented):
We can just change all AUMI/AppNames to "Chatterino", feel free to do that in this PR - I'm assuming that should not break anything. Rather that be changed now than before we start creating shortcuts

Nerixyz commented:
> Should we change the name of the App [here](https://github.com/Chatterino/chatterino2/blob/7fee3f7e375a099b5207c215389343f8f1d95bf2/src/singletons/Toasts.cpp#L231)?

This worked for me. I'd advocate for this.

---

Honestly, I now think this should be disabled by default. From what we know: installations with the installer (probably the majority) aren't affected. But we add a shortcut there by default. In Chatterino7, for example, the created shortcut is named `Chatterino7.lnk`. With this PR, an additional shortcut (`Chatterino.lnk`) would be created. But that's not necessary (could also change the app name there, I suppose). It's really only for portable users (or users of `cmake --install`).

Nerixyz commented:
> Does it show the correct name and icon for you on stable 2.5.2 installed with its installer?

Hmm, no, it doesn't. The installer changes are good, though, I'm fine with them. I guess we could enable it by default... But we should definitely change the name that we give to WinToast then. Because the installer creates `Chatterino.lnk`, not `Chatterino2.lnk`.

8thony commented:
Changes made:
- Renamed shortcut to `Chatterino`
- Renamed all AppUserModelID
- Added AppUserModelID to main app [more about this here](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-setcurrentprocessexplicitappusermodelid)
   - This makes sure the taskbar shortcut uses the same AppUserModelID.
	   - Downside: the old taskbar shortcut "break" (c2 need to be re-pinned one time)
